consensus/poet/sgx/sawtooth_poet_sgx/libpoet_enclave/poet_enclave.cpp

```diff
@@ -392,6 +394,13 @@ poet_err_t ecall_CreateSignupData(
             ret,
             "Failed to create PoET public/private key pair");
 
+        // Create the monotonic counter bound to the keypair
+        ret =
+            sgx_create_monotonic_counter(
+                &validatorSignupData.counterId,
+                &initialCounterValue);
+        sp::ThrowSgxError(ret, "Failed to create monotonic counter.");
```

Comment:
I don't see where counters are destroyed,  this code will exaust the supply of system monotonic counters once the number of ecall_CreateSignupData  is greater then the number of counters.